utils/CircuitCheck/UnitaryBuilder.cpp

```diff
@@ -53,7 +53,7 @@ LogicalResult UnitaryBuilder::build(func::FuncOp func) {
 
       for (auto &&[newQuantumOp, quantumOp] : llvm::zip(
                quake::getQuantumResults(op), quake::getQuantumOperands(op)))
-        qubitMap[newQuantumOp] = qubitMap[quantumOp];
+        qubitMap.insert({newQuantumOp, qubitMap[quantumOp]});
```

Comment:
Is the intention here to overwrite the lhs if it is already in the map? If so, then `insert` is going to be frustrating since it only does the insertion if the key is NOT in the map already.

Reply:
> Is the intention here to overwrite the lhs if it is already in the map? If so, then `insert` is going to be frustrating since it only does the insertion if the key is NOT in the map already.

No, `newQuantumOp` won't exist in `qubitMap[]` yet because we're `func.walk`-ing the block, and this is a new result coming from the operation being walked, so it should always indeed be inserting here, so I think we should be safe here.